index.bs

```diff
@@ -10651,6 +10657,155 @@ The [=remote end steps=] given |session| and |command parameters| are:
 
 </div>
 
+## The autofill Module ## {#module-autofill}
+
+The <dfn export for=modules>autofill</dfn> module contains functionality for
+saving and triggering autofill.
+
+A [=/browsing context=] has a <dfn>autofill store for testing</dfn>, an
+<a>ordered map</a>.
+
+### Definition ### {#module-autofill-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+InputCommand = (
+  autofill.save //
+  autofill.trigger
+)
+</pre>
+
+### Commands ### {#module-autofill-commands}
+
+#### The autofill.save Command ####  {#command-autofill-save}
+
+The <dfn export for=commands>autofill.save</dfn> command saves a set of field
+name values to the [=/browsing context=]'s [=autofill store for testing=].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      autofill.save = (
+        method: "autofill.save",
+        params: autofill.saveParameters
+      )
+
+      autofill.saveParameters = {
+        context: browsingContext.BrowsingContext,
+        fields: [*autofill.field]
+      }
+
+      autofill.field = (
+        name: tstr,
+        value: tstr
+      )
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl">
+     EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for autofill.save">
+
+The [=remote end steps=] with <var ignore="">session</var> and
+|command parameters| are:
+
+1. Let |context id| be the value of the <code>context</code> field of
+   |command parameters|.
+
+1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+   with |context id|.
+
+1. Let |store| be |context|'s [=autofill store for testing=].
+
+1. Let |fields| be the value of the <code>fields</code> field of |command
+   parameters|.
+
+1. For each |field| in |fields|:
+
+   1. Let |name| be |field|'s <code>name</code>.
+
+   1. Let |value| be |field|'s <code>value</code>.
+
+   1. Set |store|[|name|] to |value|.
+
+1. Return [=success=] with data null.
+
+</div>
+
+#### The autofill.trigger Command ####  {#command-autofill-trigger}
+
+The <dfn export for=commands>autofill.trigger</dfn> command triggers autofill
+on a particular form field using the [=/browsing context=]'s [=autofill
+store for testing=].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      autofill.trigger = (
+        method: "autofill.trigger",
+        params: autofill.triggerParameters
+      )
+
+      autofill.triggerParameters = {
+        context: browsingContext.BrowsingContext,
+        element: script.SharedReference
```

Comment:
Potentially, based on attributes in the storage, multiple suggestions could be available in autofill. I think the protocol should either allow defining the entire data that is being triggered or some sort of an event to notify what autofill options are available based on the storage. WDYT?

Reply:
Interesting!

Do you have in mind a `getAutofillOptions` method that would then allow the caller to call `trigger` with one of the options?

Reply:
with explicitly passing in the autofill storage for each test, do you still deem this relevant?
I'm thinking that we could make it a requirement that the keys in the autofill storage need to be unique, or alternatively say if there are multiple values for the same key, take the first one?

Unless we wanted to specifically include the option to test such a selection process, which would then be a different story. I'm not clear though if this is a useful test in this context, would appreciate some more background in this case.